src/App/WarehouseService.cs

```diff
@@ -3,25 +3,45 @@
 
 public class WarehouseService
 {
-    private readonly string _connectionString = "Host=localhost;Username=postgres;Password=yourpassword;Database=warehouse";
+    private readonly string _connectionString = Environment.GetEnvironmentVariable("CONNECTION_STRING");
 
-    public void SaveData(string key, string value)
+    public void SaveData(string partName, int bikeTypeId, int stock)
     {
+        try
+        {
+            using var connection = new NpgsqlConnection(_connectionString);
```

Comment:
[The data source concept is new in Npgsql 7.0, and is the recommended way to use Npgsql.](https://www.npgsql.org/doc/basic-usage.html) 

in other words, fine for this project, but in the future see this as deprecated